### PR TITLE
Add sigmoidf1_loss and softf1_loss

### PR DESCRIPTION
**Pull request by old-school-kid**
# Description
Implementation of the paper [sigmoidF1: A Smooth F1 Score Surrogate Loss for Multilabel Classification](https://arxiv.org/pdf/2108.10566v1.pdf)

Brief Description of the PR:
Adds 2 new loss function sigmoidf1_loss and softf1_loss. Both of the loss functions are based on f1 accuracy and have been found to work better than their counterparts in multi-class and multi-label classification. 
Unit test for both the losses has been added.

The sigmoidf1_loss function along with its unit testing can also be done in this [notebook](https://colab.research.google.com/drive/1nCY0oFx8k7x4Mz-KmTBjY4dPlP-GnGl-?usp=sharing)
The softf1_loss function along with its unit testing can also be done in this [notebook](https://colab.research.google.com/drive/14n1n96YZMIh58YDuGu4OOz7ilTNQNzlM?usp=sharing)

Fixes # (issue)
#2561 
## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [x] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [x] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?
I have cloned it ran it on my local machine. Further scope for tests has been provided in form of colab notebooks above.
If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  check on various parameters
* check on different shape and datatype
* check different size of y_pred and y_true
* check for possible arithmetical errors including ZeroDivison, InfiniteMultiplication, InfiniteSummation etc

### CODE REVIEW

**Comment by old-school-kid** [comment]
@WindQAQ would you please look into this?
Thanks
